src/Provider/AbstractProvider.php

```diff
@@ -37,8 +37,9 @@
     /**
      * @param HttpClient|null     $httpClient
      * @param RequestFactory|null $requestFactory
+     * @param array               $options
      */
-    public function __construct(HttpClient $httpClient = null, RequestFactory $requestFactory = null)
+    public function __construct(HttpClient $httpClient = null, RequestFactory $requestFactory = null, array $options = [])
```

Comment:
Thanks :) Sure, I will let you know.

I'm trying to implement historical rates and the missing features while making the architecture more robust and easier to extend for 3.x versions.

I plan to release a 3.0 beta for a week or so to make sure we have all needed features and it's well tested before releasing.